hybrasyl/Messaging/Admin.cs

```diff
@@ -0,0 +1,410 @@
+                    }
+                }
+                else if (args[0].ToLower() == "enable")
+                {
+                    script.Disabled = false;
+                    return Success($"Script {script.Name}: enabled.");
+                }
+                else if (args[0].ToLower() == "disable")
+                {
+                    script.Disabled = true;
+                    return Success($"Script {script.Name}: disabled.");
+                }
+                else if (args[0].ToLower() == "status")
+                {
+                    var scriptStatus = string.Format("{0}:", script.Name);
+                    string errorSummary = "--- Error Summary ---\n";
+
+                    if (script.LastRuntimeError == string.Empty &&
+                        script.CompilationError == string.Empty)
+                        errorSummary = string.Format("{0} no errors", errorSummary);
+                    else
+                    {
+                        if (script.CompilationError != string.Empty)
+                            errorSummary = string.Format("{0} compilation error: {1}", errorSummary,
+                                script.CompilationError);
+                        if (script.LastRuntimeError != string.Empty)
+                            errorSummary = string.Format("{0} runtime error: {1}", errorSummary,
+                                script.LastRuntimeError);
+                    }
+
+                    // Report to the end user
+                    return Success($"{scriptStatus}\n\n{errorSummary}", MessageTypes.SLATE_WITH_SCROLLBAR);
+                }
+            }
+            return Fail("Script {args[1].Trim()}: not found.");
+        }
+    }
+
+    class RollcharCommand : ChatCommand
+    {
+        public new static string Command = "rollchar";
+        public new static string ArgumentText = "<string class> <byte level>";
+        public new static string HelpText = "Simulate a new character with the specified class and level, and show resulting HP/MP.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (!Enum.IsDefined(typeof(Enums.Class), args[1]))
+                return Fail($"Invalid class {args[0]}");
+
+            if (!byte.TryParse(args[2], out byte level) || level < 1 || level > Constants.MAX_LEVEL)
+                return Fail("Invalid level: must be between 1 and {Constants.MAX_LEVEL}");
+
+            // Create a fake User, and level them up to the desired level
+
+            var testUser = new User();
+            testUser.Map = new Map();
+            testUser.Stats.BaseHp = 50;
+            testUser.Stats.BaseMp = 50;
+            testUser.Class = (Enums.Class)Enum.Parse(typeof(Enums.Class), args[1]);
+            testUser.Stats.Level = 1;
+
+            while (testUser.Stats.Level < level)
+            {
+                testUser.GiveExperience(testUser.ExpToLevel);
```

Comment:
testUser doesn't have a Client instantiated, so this call results in NullREferenceException.

Reply:
this is a slash command that was added by a previous contributor that I just ported over - I'll look closer at this

Reply:
I just went ahead and removed this.